src/engine/Server.cpp

```diff
@@ -456,16 +458,25 @@ CPP_template_2(typename RequestT, typename ResponseT)(
                                 parsedOperation._originalString)));
     }
     if (parsedOperation.hasUpdateClause()) {
-      co_return co_await processUpdate(
-          std::move(parsedOperation), requestTimer, cancellationHandle, qec,
-          std::move(request), send, timeLimit.value());
+      AD_CORRECTNESS_CHECK(parsedOperation.hasUpdateClause());
```

Comment:
This check pretty much redundant here. You just checked this condition in the if clause. You can just remove it.